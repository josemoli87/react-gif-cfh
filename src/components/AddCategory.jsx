import {useState} from 'react';

// eslint-disable-next-line react/prop-types
export const AddCategory = ({onNewCategory}) => {
	const [inputValue, setInputValue] = useState('');

	const onInputChange = ({target}) => {
		setInputValue(target.value);
	};

	const onSubmit = (event) => {
		event.preventDefault();
		if (inputValue.trim().length <= 1) return;

		setInputValue('');
		onNewCategory(inputValue.trim());
	};

	return (
		<form onSubmit={onSubmit}>
			<input
				type='text'
				placeholder={inputValue}
				value={inputValue}
				onChange={onInputChange}
			/>
		</form>
	);
};
