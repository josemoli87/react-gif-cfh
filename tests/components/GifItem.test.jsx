import {render} from '@testing-library/react';
import {GifItem} from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {
	const title = 'Goku';

	const url = 'https://DBZ.com/goku.jpg';

	test('debe hacer match con el snapshot ', () => {
		const {container} = render(<GifItem title={title} url={url} />);
		expect(container).toMatchSnapshot();
	});
});
