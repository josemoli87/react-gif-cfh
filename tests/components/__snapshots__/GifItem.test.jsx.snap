// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://DBZ.com/goku.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;

exports[`Pruebas en <GifItem/> debe hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="url"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
